### ISSUE TITLE
Performance regression between 1.5.3.9 and 1.7.1.2

### ISSUE THREAD

**Issue by buttercookie42** [leezer3/OpenBVE#422]
### Description

After updating OpenBVE from 1.5.3.9 to 1.7.1.2, I've noticed that the frame rate on the Jubilee Line has decreased quite noticeably, and even the new experimental renderer is still slower than the old version.

### Reproduction

Using the default settings except for switching to full screen mode and 1920x1080 resolution, launching "Jubilee Line V2 Stratford - Waterloo.csv" with the default train included with the route and then turning on the FPS counter, I get
* 1.5.3.9: 27 - 28 fps
* 1.7.1.2: 7 fps (!)
* 1.7.1.2 (new renderer): 20 - 23 fps

## Route

http://brnobve.eu/?p=4242

## Train

http://brnobve.eu/?p=4242

## Logs

[log.txt](https://github.com/leezer3/OpenBVE/files/4061450/log.txt)

### Related information 

* Windows 10 1909, Core i5-3570, AMD Radeon HD 7850
* Keyboard control

**Comment by leezer3** [comment]
Ouch, that's pretty nasty :(

I'll admit I haven't so much been focusing on the old renderer, but as to getting the new renderer into better shape. I'm also working rather too much at the minute, which means I don't have as much time as I want to deal with things.
(Essentially speaking, the more animated objects / train parts there are, the better it performs)

I've found another boost today, which gets the new renderer very close to the old with the Jubilee (although not quite there yet....), and I'm still going after the performance figures for this.
Might take a little longer, but we'll get it to as fast or better in all circumstances soon.

Having said that, the old renderer obviously needs a little TLC to get the speed back again. Will try and do something about that soon.

**Comment by buttercookie42** [comment]
I didn't perform any extensive testing, but a quick spot check with my personal settings and the North London Line seems to indicate that while the difference between 1.7.1.2 and 1.7.2.0 isn't very large, on my system the new renderer now seems to have actually caught up with (or possibly even overtaken, though only by a very small margin) the performance of the old renderer in 1.5.3.9 (or even 1.7.0.3 for that matter).

**Comment by Kenny-Hui** [comment]
On dedicated GPU, I didn't notice much difference between 1.5 and 1.7 (Both with old render, and I've even doubled my fps with the new render)

But on an iGPU (Intel HD 4600) machine, it seems the performance has decreased over time.
On 1.5, I was able to achieve 25-30fps on an Light Rail Route (This performance decrease really applies to all addon, I only used LRT for testing)
On 1.7.2.1 (Old Render), It's about 10-15fps
Even with the new render enabled, it's still around 22-25 fps.

Many lower-end player also reported slow route loading & extreme fps drop on newer versions, and they're unwilling to switch to newer versions (Making them not being able to use new features. At the same time content creator has to put more effort on backward compatibility)
